### ISSUE TITLE
Getting Nan's on model-parallel 

### ISSUE THREAD

**Issue by adizhol** [facebookresearch/maskrcnn-benchmark#1216]
Hi,

I've been trying to train a maskcrnn-benchmark with backbones of Resnet-FPN and ResneXt-FPN on multiple GPUs (V100) using model-parallel approach as described in https://pytorch.org/tutorials/intermediate/model_parallel_tutorial.html.

After a few iterations both model encounter Nan's in backbone output.
This problem doesn't occur when training on a single GPU (same learning rate).

pytorch 1.2 and 1.4
cuda 10.1.168

Has anyone encountered this problem?

I don't have a public repo, but basically, my code is like this:
initializing trainer:

    device_ids = cfg.MODEL.DEVICE.split(',') # cuda:0,cuda:1
    model = build_detection_model(cfg)
    if len(device_ids) > 1: #keep backbone in device 0. The rest go to device 1
        model.to(device_ids[1])
        model.backbone.body = model.backbone.body.to(device_ids[0])
        model.backbone.fpn = model.backbone.fpn.to(device_ids[1])
        model.rpn = model.rpn.to(device_ids[1])
    else:
        model.to(device)
The forward of maskrcnn:

        if len(self.devices) > 1:
            resnet_out = self.backbone.body(images.tensors) #inputs are in device 0
            features = self.backbone.fpn([x.to(self.devices[1]) for x in resnet_out])
        else:
            features = self.backbone(images.tensors)
        proposals, proposal_losses = self.rpn(images, features, targets)
        if self.roi_heads:
            x, result, detector_losses = self.roi_heads(features, proposals, targets) #non Nans up to here
...
There are no Nans until calling the roi_head

update: The Nan's are from the poolers, probably from ROIAlign...
Any idea if there special care needs to be taken in .cu code to support model-parallel?

this issue on pytorch github:
https://github.com/pytorch/pytorch/issues/34134

**Comment by adizhol** [comment]
solved by replacing ROIALIGN with torchvision ROIAlign
